### ISSUE TITLE
What is the way to shut down turnserver by Unix signal?

### ISSUE THREAD

**Issue by atuk721** [coturn/coturn#945]
A turnserver seems it can only be shut down by SIGKILL signal.
Is there an other way to shut down it?
Or sending SIGKILL is an intended way to shut down?

I've checked it by the following Docker images.

* coturn/coturn:4.5.2-r14-alpine
* coturn/coturn:4.5.2-r14-debian

**Comment by eakraly** [comment]
Yes, this is a common practice.
SIGKILL or SIGTERM - both will cause turnserver to shutdown

**Comment by atuk721** [comment]
@eakraly 
I cannot kill a turnserver sending SIGTERM by the following method. Do I have any mistake?

```sh
docker run --rm -d --name coturn coturn/coturn

# I cannot kill it.
docker kill -s TERM coturn

# I can kill it.
docker kill -s KILL coturn
```

**Comment by eakraly** [comment]
You are right...
I do not see any support for SIGTERM in the code, and SIGKILL obviously kills the process

Do you think there should be support for SIGTERM (for gracious shutdown for example)?

**Comment by ggarber** [comment]
There is a "shutdown" command : https://github.com/coturn/coturn/blob/master/src/apps/relay/turn_admin_server.c#L978

I have never used it but it looks like what you need so you could give it a try.

**Comment by atuk721** [comment]
Thank you for your information.
I think this should be support for SIGTERM or should have `STOPSIGNAL SIGKILL` in the Dockerfile.

One example of a current problem is using on Kubernetes.
A termination process of Pod on Kubernetes starts to send SIGTERM and then wait to stop it 30 seconds by default.
If it doesn't be stopped, Kubernetes sends SIGKILL to it.

https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-termination

So now we need to wait a long time to stop or restart it although we can change `terminationGracePeriodSeconds` for Pod, not Container.

**Comment by jonesmz** [comment]
I'm looking for a way to cause the turnserver to stop accepting new allocations, but continue servicing already established allocations, including adding new permissions to the existing allocations.

Ideally this can be done with a unix signal (e.g. sighup or similar), without having to use the telnet admin console.

The use case for this is to support shutting down a coturn instance gracefully without terminating any existing connections. Eventually the existing connections will disconnect on their own. This would be done for example when a new build is being deployed to the infrastructure hosting coturn. The existing process would be taken out of the load balancer, but not shut down, so that live connections can continue while any new connections are sent to the new instances.

**Comment by eakraly** [comment]
@jonesmz No, there is no implementation for that and it is a good thing to have it.
There is a handled for SIGTERM but it does exactly the same thing for now (stop listener)
